### PR TITLE
Use modern php versions

### PR DESCRIPTION
**Pull request by vkhramtsov**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #28 
| License       | MIT

### CODE REVIEW

**Comment by vkhramtsov** [comment]
@sstok Would you be do kind to review this PR?

**Comment by sstok** [comment]
@stof I know Symfony considers dropping support for older PHP and Symfony versions a BC break, do you consider it acceptable here? Or would this require a major release.

Personally I don't mind dropping support for Symfony 2 as current API is already forward compatible and Symfony 2 is EOL. And anyone stuck on a older version can still use the older versions of this library.

**Comment by vkhramtsov** [comment]
@sstok After taking decision and merging (possibly) this PR, I'm going to do the same for bundle

**Comment by stof** [comment]
> I know Symfony considers dropping support for older PHP and Symfony versions a BC break, do you consider it acceptable here? Or would this require a major release.

As long as your composer.json properly defines the constraints, that's not a BC break.

Symfony decided to bump min PHP versions only in major versions due to other reasons (allowing predictable bumps there too), not due to it being a BC break.

I would say that dropping support for older versions of deps puts some expectations on the maintenance of an older version of the lib:
- if that older version is still maintained, users might still expect to get bugfixes in a version of your lib supporting it
- if that older version is EOL, users might still expect a backport of your own security patches for *some* amount of time after you drop support, depending on the ease to upgrade.

These expectations means that you should at least use a minor version for such bump (to still be able to make some patch releases in the previous minor). But nothing forces using a major version here.

File: .travis.yml [hunk 1]
```diff
@@ -2,13 +2,18 @@ language: php
...
         - php: 7.1
           env: DEPENDENCIES='low'
+        - php: 7.2
+          env: DEPENDENCIES='dev' lint=1
+        - php: 7.2
+          env: DEPENDENCIES='low'
+        - php: 7.3
+          env: DEPENDENCIES='dev' lint=1
+        - php: 7.3
+          env: DEPENDENCIES='low'
```
Comment (stof) [inline .travis.yml]: running `DEPENDENCIES='low'` and `DEPENDENCIES='dev'` looks useless to me. It increases the Travis test matrix (making it more likely to reach the maximum allowed concurrency).

And in this setup, there is no job running unmodified deps anymore btw (as 5.6 and 7.0 were the one doing it before)

File: .travis.yml [hunk 2]
```diff
@@ -2,13 +2,18 @@ language: php
...
     include:
-        - php: 5.6
-        - php: 7.0
-          env: lint=1
         - php: 7.1
           env: DEPENDENCIES='dev' lint=1
         - php: 7.1
           env: DEPENDENCIES='low'
+        - php: 7.2
+          env: DEPENDENCIES='dev' lint=1
```
Comment (stof) [inline .travis.yml]: there is no reason to have multiple jobs performing the CS linting. That linting does not depend on the PHP version.

File: .travis.yml (hunk 1 above)
Comment (vkhramtsov) [reply .travis.yml]: I think that `low` and `dev` help us to test it better.
Should I add build with unmodified deps for every php version?
@sstok What do you think?

Comment (stof) [reply .travis.yml]: well, having 1 job for low deps is useful to ensure that lowest bounds are OK.
having 1 job for dev deps is useful to catch issues with upcoming version (getting a chance to fix them before they reach a release).
But the jobs using deps without modifying them are the most important ones: that's how the package will be used by most users (a dev deps is used only if you explicitly allow it in composer, and a lower dep is used only if some conflict forbids using the latest release)

Comment (vkhramtsov) [reply .travis.yml]: Updated

File: .travis.yml (hunk 2 above)
Comment (vkhramtsov) [reply .travis.yml]: Updated

**Comment by vkhramtsov** [comment]
@stof Would you be so kind to review this once again?

File: .travis.yml (hunk 1 above)
Comment (stof) [reply .travis.yml]: you still have lots of tests with modified deps (meaning that a single build of this matrix already outweights the concurrency quota of the github organization, and has to wait on itself to run)

Comment (vkhramtsov) [reply .travis.yml]: So what do you propose? Leave dev and low only for one php version (for example for php 7.2 ) and remove dev and low for other versions?

Comment (stof) [reply .travis.yml]: yes, exactly.

Comment (stof) [reply .travis.yml]: and this is how it is currently in master btw: only one job for each of them.

Comment (vkhramtsov) [reply .travis.yml]: Updated

Comment (vkhramtsov) [reply .travis.yml]: @stof Is it OK now?

Comment (vkhramtsov) [reply .travis.yml]: @stof Do you have more suggestions?

**Comment by vkhramtsov** [comment]
@stof @sstok Any news here?

**Comment by stof** [comment]
@vkhramtsov this is fine with me. But I'm not a maintainer of this library, so you still need to have @sstok looking at it to merge it.

**Comment by sstok** [comment]
Thank you @vkhramtsov

**Pull request merged**